packages/ensjs/src/utils/wildcardWriting.ts

```diff
@@ -0,0 +1,286 @@
+  TChain extends ChainWithEns,
+  TAccount extends Account | undefined,
+>(
+  wallet: WalletClientWithAccount<Transport, TChain, TAccount>,
+  errorResult: ErrorResult,
+  encodedName: Hex,
+  calldata: Hex,
+  account: Address,
+  expiry?: bigint,
+): Promise<Hash | undefined> {
+  const currentChain = wallet.chain
+
+  if (errorResult.errorName === 'OperationHandledOffchain') {
+    const [domain, url, message] = errorResult.args as [
+      DomainData,
+      string,
+      MessageData,
+    ]
+
+    const signature = await wallet.signTypedData({
+      account,
+      domain,
+      message,
+      primaryType: 'Message',
+      types: {
+        Message: [
+          { name: 'data', type: 'bytes' },
+          { name: 'sender', type: 'address' },
+          { name: 'expirationTimestamp', type: 'uint256' },
+        ],
+      },
+    })
+    await ccipRequest({
+      data: message.data,
+      signature: { message, domain, signature },
+      sender: message.sender,
+      urls: [url],
+    })
+  }
+
+  if (errorResult.errorName === 'OperationHandledOnchain') {
+    try {
+      const [chainId, contractAddress] = errorResult.args as [
+        bigint,
+        `0x${string}`,
+      ]
+
+      if (
+        wallet.chain.id !== chains.localhost.id &&
+        wallet.chain.id !== Number(chainId)
+      ) {
+        await wallet.switchChain({ id: Number(chainId) })
+        const chain = getChain(Number(chainId))
+        if (!chain) throw new Error('Chain not found')
+        wallet.chain = chain as TChain
+      }
+
+      let value = 0n
+      if (calldata.slice(0, 10) === WILDCARD_WRITING_REGISTER_SELECTOR) {
+        const registerParams = (await readContract(wallet, {
+          address: contractAddress,
+          abi: offchainRegisterSnippet,
+          functionName: 'registerParams',
+          args: [encodedName, expiry],
+        })) as {
+          price: bigint
+          commitTime: bigint
+          extraData: Hex
+          available: boolean
+          token: Hex
+        }
+
+        if (!registerParams.available) {
+          throw new SubnameUnavailableError(encodedName)
+        }
+
+        value = registerParams.price
+      }
+
+      await sendTransaction(wallet, {
+        account,
+        to: contractAddress,
+        value,
+        data: calldata,
+        authorizationList: [],
+      } as SendTransactionParameters<TChain, TAccount>)
+    } finally {
+      if (wallet.chain.id !== chains.localhost.id) {
+        await wallet.switchChain({ id: currentChain.id })
+        wallet.chain = currentChain
+      }
+    }
+  }
+
+  // random ethereum transaction hashes had to be returned to avoid breaking changes
+  return currentChain.id === chains.sepolia.id
+    ? '0x1d4cca15a7f535724328cce2ba2c857b158c940aeffb3c3b4a035645da697b25' // random successful sepolia tx hash
+    : '0xd4a47f4ff92e1bb213a6f733dc531d1baf4d3e439229bf184aa90b39d2bdb26b' // random successful mainnet tx hash
+}
```

Comment:
I see that it is necessary but for some reason it bothers me huehuehuehe

Reply:
it does feel wrong, but it was the only way we found to keep backward compatibility. I'm looking forward to their feedback of a better way to handle this scenario